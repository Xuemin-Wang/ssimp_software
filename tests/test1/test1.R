
setwd("/Users/admin/Documents/Work/Projects/tagging/ssimp_software/ref/")
library(readr)

## ref
vcf.raw <- read_tsv("1000genomes_90snps_100samples.vcf")
snps <- vcf.raw$ID

## prep vcf
map <- vcf.raw[,1:5]
G.char <- t(as.data.frame(vcf.raw[,-(1:9)]))
G <- matrix(NA, ncol = ncol(G.char), nrow = nrow(G.char))
G[G.char == "0|0"] <- 0
G[G.char == "0|1"] <- 1
G[G.char == "1|0"] <- 1
G[G.char == "1|1"] <- 2
colnames(G) <- map$ID

## gwas
path.gwas <- "/Users/admin/Documents/Work/Projects/tagging/ssimp_software/gwas/"
gwas <- read_tsv(paste0(path.gwas, "gwas_example.txt"))

## snps2impute
snps2imp <- setdiff(snps, gwas$SNP)
tsnps <- intersect(snps, gwas$SNP)

## imputation


## prep C.mat
v.g <- apply(G, 2, var)
C.mat <- cor(G[,v.g > 0], use = "pairwise")
lambda <- 2/sqrt(nrow(G))

C.mat <- (C.mat * (1-lambda)) + diag(C.mat) * lambda

tsnps <- intersect(tsnps, colnames(C.mat))
snps2imp <- intersect(snps2imp, colnames(C.mat))

rho <- C.mat[tsnps, snps2imp]
C <- C.mat[tsnps, tsnps]

## gwas data
gwas.sub <- gwas[gwas$SNP %in% tsnps, ]
Z <- gwas.sub$BETA/gwas.sub$SE


## actual imputaton
res <- t(rho) %*% solve(C) %*% Z
out <- data.frame(SNP = rownames(res), Z.imp = res)

write_tsv(out, "test1.R.out")
