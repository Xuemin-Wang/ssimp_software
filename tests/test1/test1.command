ssimp --data ../../gwas/gwas.txt --refpanel EUR,EAS`
