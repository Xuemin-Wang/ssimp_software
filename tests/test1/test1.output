Expected output
will generate a `gwas.imp.log` file and a `gwas.imp.out` file.
