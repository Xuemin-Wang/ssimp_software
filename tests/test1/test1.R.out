SNP	Z.imp
rs41297828	-1.0970995277031692
rs570570299	0.0034757065236227236
rs5763214	0.5498316407041726
rs192341766	0.21520811574770538
rs66715066	-0.7546519641626462
rs568553147	-0.9146111185067909
rs534916624	-0.6719641520339276
rs79504	-1.700385778263511
rs79505	-1.6985397152533102
